src/Spago/Psa.purs

```diff
@@ -60,9 +60,12 @@ psaCompile globs pursArgs psaArgs = do
 
         pure $ FO.isEmpty out'.stats.allErrors
 
-  if Array.all identity arrErrorsIsEmpty then do
+  if Array.all identity arrErrorsIsEmpty && Cmd.exitedOk result then do
```

Comment:
I don't think checking for an exit code of 0 is the right thing to do here - when compilation fails because of errors in the build `purs` will exit with `1`.
Let's try to catch this earlier - e.g. I think we should be able to prevent the JSON decoding error: let's try to run the json parsing separately, and print a sensible error there